### PR TITLE
Update samples to use okta spring boot starter and spring boot 3

### PR DESCRIPTION
**Pull request by jimmyjames**
## Changes

Updates the servlet and webflux samples to:

* Use the Okta Spring Boot Starter
* Update to Spring Boot 3

A corresponding quickstart article change will also be made.

### CODE REVIEW

File: 01-Authorization-MVC/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-bin.zip
```
Comment (mraible) [inline 01-Authorization-MVC/gradle/wrapper/gradle-wrapper.properties:3]: 8.4 is the latest version and would be better for Java 21 compatibility. https://docs.gradle.org/8.4/release-notes.html

```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-bin.zip
```

File: 01-Authorization-MVC/src/main/java/com/auth0/example/security/SecurityConfig.java [hunk 2]
```diff
@@ -1,58 +1,37 @@
...
-import org.springframework.security.oauth2.jwt.*;
 import org.springframework.security.web.SecurityFilterChain;
 
+import static org.springframework.security.config.Customizer.withDefaults;
+
 /**
  * Configures our application with Spring Security to restrict access to our API endpoints.
  */
+@Configuration
 @EnableWebSecurity
```
Comment (mraible) [inline 01-Authorization-MVC/src/main/java/com/auth0/example/security/SecurityConfig.java:16]: `@EnableWebSecurity` is not needed if it's a Spring Boot app. It's only necessary for Spring apps w/o Boot.

File: 01-Authorization-MVC/build.gradle [hunk 3]
```diff
@@ -6,21 +6,24 @@ buildscript {
 
 plugins {
     id 'java'
-    id 'org.springframework.boot' version '2.7.0'
-    id 'io.spring.dependency-management' version '1.1.0'
+    id 'org.springframework.boot' version '3.1.4'
```
Comment (frederikprijck) [inline 01-Authorization-MVC/build.gradle:9]:
```suggestion
    id 'org.springframework.boot' version '3.1.5'
```

File: 01-Authorization-WebFlux/build.gradle [hunk 4]
```diff
@@ -6,19 +6,22 @@ buildscript {
 
 plugins {
     id 'java'
-    id 'org.springframework.boot' version '2.5.12'
-    id 'io.spring.dependency-management' version '1.0.9.RELEASE'
+    id 'org.springframework.boot' version '3.1.4'
```
Comment (frederikprijck) [inline 01-Authorization-WebFlux/build.gradle:9]:
```suggestion
    id 'org.springframework.boot' version '3.1.5'
```

**Review by frederikprijck — DISMISSED** [review]
Just added some comments to align with comments made elsewhere about springboot version, can go without 👍

**Comment by jimmyjames** [comment]
Think we will need https://github.com/auth0-samples/api-quickstarts-tests/pull/6 for the API quickstarts test to run.

**Review by poovamraj — APPROVED** [review]

**Pull request merged**